lib/ComponentsManager.ts

```diff
@@ -95,6 +100,81 @@ export class ComponentsManager<Instance> {
     }
     return <Error> error;
   }
+
+  // TODO: put this in separate file
+  protected findOverrides(): Record<string, Record<string, Resource>> {
```

Comment:
I thought about the `ConfigRegistry`, but if you first register the config with the override and only then the config with the resource it might cause issues. Or first a config with override 2, then one with the resource, and then the one with override 1.

Perhaps if we iterate over all the overrides found already every time a new config gets added.

Reply:
The problem with the `configConstructorPool` is that the `initialize` function is called multiple times recursively and this is something that only needs to happen once, once initialization starts, so it's not clear to me yet where it would go in there.

Could change the `configConstructorPool` interface so it has an `initialize` and an `initializeResource` function. Where `initialize` first determines the overrides, and then calls `initializeResource` (and all other recursive calls of `initialize` get changed to `initializeResource`).

Or have an `OverrideConfigConstructorPool` that wraps around the original implementation.

Reply:
`instantiate` could also be called multiple times for that matter. So we'll probably need to do some caching, and work with a flag to mark that overriding already took place.

Reply:
Is it possible that new configs get added inbetween 2 `instantiate` calls? If yes we would have to cache based on the `configRegistry` state somehow

Reply:
> Is it possible that new configs get added inbetween 2 instantiate calls?

In theory, yes. But it's not how it _should_ be done.

> If yes we would have to cache based on the configRegistry state somehow

It might be easier to just provide an explicit way to reset/regenerate overrides when the user decides to register a new config after already having instantiated one before.
Because I think caching on configRegistry state would become too complex.